lib/katello_attach_subscription/utils.rb

```diff
@@ -14,6 +14,25 @@ def self.search_args(search)
         search
       end
     end
+
+    def self.merge_subs(current_sub, sub_to_merge, command)
+      if not current_sub.nil?
+        current = current_sub.clone
+      else
+        current = {}
+      end
+      new_sub = sub_to_merge.clone
+      case command
+      when "override"
+        current = new_sub
+      else
+        if current.nil?
+          current = new_sub
+        else
+          current.merge!(new_sub)
+        end
+      end
+      return current.clone
```

Comment:
why returning a clone and not `current` itself?

Reply:
because from tests, I notice that, for one host, if we merge hashes taken from different entry of parsed_sub, also parsed_sub is being updated, that it's an issue for the next host entry